### PR TITLE
KAFKA-14262: Deletion of MirrorMaker v1 deprecated classes & tests

### PR DESCRIPTION
**Pull request by abhi-ksolves**
This PR removes the Mirror Maker v1 classes and tests, as they have been deprecated according to [KIP-720](https://cwiki.apache.org/confluence/display/KAFKA/KIP-720%3A+Deprecate+MirrorMaker+v1) 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -845,7 +845,7 @@ <h5><a id="upgrade_300_notable" href="#upgrade_300_notable">Notable changes in 3
         <code>DefaultReplicationPolicy</code>, cannot prevent replication cycles based on topic names, so take care to avoid cycles when constructing your
         replication topology.
     </li>
-    <li> The original MirrorMaker (MM1) and related classes have been deprecated. Please use the Connect-based
+    <li> The original MirrorMaker (MM1) and related classes have been removed. Please use the Connect-based
```
Comment (chia7712) [inline docs/upgrade.html]: This doc is related to 3.0.0 release, and hence we should not change it. Could you please add new entry for 4.0.0 release?

**Review by chia7712 — COMMENTED** [review]
@abhi-ksolves thanks for this patch. I will take another look later

File: docs/upgrade.html (hunk 1 above)
Comment (abhi-ksolves) [reply docs/upgrade.html]: @chia7712 I have added the entry. Please review it to ensure its correctness. The entry has been added as follows: `<h4><a id="upgrade_4_0_0" href="#upgrade_4_0_0">Upgrading to 4.0.0 from any version 0.8.x through 3.9.x</a></h4>
    <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4.0.0</a></h5>`

**Comment by chia7712** [comment]
@abhi-ksolves could you please remove related scripts and e2e?

https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/mirror_maker.py
https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/core/mirror_maker_test.py
https://github.com/apache/kafka/blob/trunk/bin/kafka-mirror-maker.sh
https://github.com/apache/kafka/blob/trunk/bin/windows/kafka-mirror-maker.bat

**Comment by abhi-ksolves** [comment]
Hi @chia7712 
Mentioned scripts & tests files are removed.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**